art/sheetMusicGenerator-s1072489/index.js

```diff
@@ -0,0 +1,177 @@
+  
+  const newClefPointsTop = clefPointsTop.map(clef_point => [clef_point[0] + movementx, clef_point[1] + movementy - 577*scale + 243*scale+ 4.5]); // Manual 4.5 extra px
+  const newClefPointsBottom = clefPointsBottom.map(clef_point => [clef_point[0] + movementx, clef_point[1] + movementy]);
+  
+
+  drawLines([newClefPointsTop], {fill: "black"});
+  drawLines([newClefPointsBottom], {fill: "black"});
+}
+
+if (clef == "treble") {
+  drawTrebleClef(9, 18, 0.05);
+}
+
+// Draw barlines
+finalLines.push([[10, 15], [10, 35]]);
+finalLines.push([[72, 15], [72, 35]]);
+finalLines.push([[114, 15], [114, 35]]);
+finalLines.push([[115, 15], [115, 35]]);
+drawLines([[[115, 15], [115, 35], [115.5, 35], [115.5, 15]]], {fill: "black"});
+
+// Function to get note y-positions
+function getNoteY(noteLine) {
+  const gapHeight = 5;
+  return 35 - (noteLine - 1) * gapHeight;
+}
+
+// Function to draw an oval note
+function drawOval(xPos, yPos, radiusX, radiusY, isFilled) {
+  const ovalPoints = [];
+  const numPoints = 20;
+  for (let i = 0; i < numPoints; i++) {
+    const angle = (Math.PI * 2 * i) / numPoints;
+    const x = xPos + radiusX * Math.cos(angle);
+    const y = yPos + radiusY * Math.sin(angle);
+    ovalPoints.push([x, y]);
+  }
+  // Close the oval by connecting the last point to the first
+  ovalPoints.push(ovalPoints[0]);
+  
+  // Draw the oval (filled or not)
+  if (isFilled) {
+    drawLines([ovalPoints], {fill: "black"}); // fill the oval for crotchets (black)
+  } else {
+    drawLines([ovalPoints]); // draw the oval for minims (unfilled)
+  }
+  
+  return ovalPoints;
+}
+
+// Determine stem direction based on position and previous stem direction
+function getStemDirection(noteY, previousDirection) {
+  const middleLineY = getNoteY(3); // Line 3 is the middle of the staff
+  if (noteY < middleLineY) {
+    return "up";
+  } else if (noteY > middleLineY) {
+    return "down";
+  } else {
+    return previousDirection; // If it's on the middle line, follow the previous note's stem direction
+  }
+}
+
+// Draw the notes based on the input
+let previousStemDirection = "up"; // Default starting stem direction
+musicInput.forEach((bar, barIndex) => {
+  bar.forEach((note, noteIndex) => {
+    const [noteLine, noteLength] = note;
+    const xPos = (barIndex * 40) + (noteIndex * 12) + 40;
+    const yPos = getNoteY(noteLine);
+
+    // Draw the note
+    const isFilled = (noteLength === 1);
+    drawOval(xPos, yPos, 4, 2, isFilled);
+
+    // Determine stem direction
+    const stemDirection = getStemDirection(yPos, previousStemDirection);
+    previousStemDirection = stemDirection;
+
+    // Draw the stem
+    if (stemDirection === "down") {
+      finalLines.push([[xPos + 4, yPos], [xPos + 4, yPos - 13]]);
```

Comment:
-4?

Reply:
I have a damn licentiate in music and I get it wrong... TWICEEEEEEEEE